lib/rules/no-await-in-condition.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const esquery = require('esquery');
+
+const findInExpressionTree = require('../util/find-in-expression-tree');
+const isNodeChaiCall = require('../util/is-node-chai-call');
+const isNodeChaiAsPromised = require('../util/is-node-chai-as-promised');
+
+function checkHasAwait (context, node) {
+  const [
+    chaiCall,
+    chaiAsPromisedCall
+  ] = findInExpressionTree(node, [
+    isNodeChaiCall, isNodeChaiAsPromised
+  ]);
+
+  if (chaiAsPromisedCall &&
+    chaiCall && esquery(chaiCall.parent, '.arguments[type="AwaitExpression"]')
+  ) {
+    context.report({
+      node: chaiAsPromisedCall,
+      message: 'Uses of chai-as-promised must not occur with an inner await',
+    });
+    return true;
+  }
+}
```

Comment:
Please add this to ensure the function has a consistent return-type

```suggestion
  }
  return false;
}
```